x-pack/docs/en/rest-api/security/update-api-key.asciidoc

```diff
@@ -1,5 +1,270 @@
 [role="xpack"]
 [[security-api-update-api-key]]
-=== Update API key information API
+=== Update API key API
 
-coming::[8.4.0]
+++++
+<titleabbrev>Update API key</titleabbrev>
+++++
+
+[[security-api-update-api-key-request]]
+==== {api-request-title}
+
+`PUT /_security/api_key/<id>`
+
+[[security-api-update-api-key-prereqs]]
+==== {api-prereq-title}
+
+* To use this API, you must have at least the `manage_own_api_key` cluster privilege.
+Users can only update their own API keys, i.e., API keys they created.
+To update another user's API key, you must use the run as feature.
+For more information, see <<run-as-privilege>>.
+
+IMPORTANT: It's not possible to use an API key as the authentication credential for this API.
+To update an API key, user password or <<security-api-get-token, token credentials>> are required.
+
+[[security-api-update-api-key-desc]]
+==== {api-description-title}
+
+The API keys created by <<security-api-create-api-key,create API Key>> or <<security-api-grant-api-key,grant API Key>> can be updated using this API.
+It's not possible to update expired API keys, or API keys that have been invalidated by <<security-api-invalidate-api-key,invalidate API Key>>.
+
+This API supports updates to an API key's access scope, i.e., its permissions, and its metadata.
+Other properties such as `name` or `expiration` cannot be updated.
+
+IMPORTANT: An API key's permissions are always restricted by the permissions of the authenticated user creating or updating the API key.
```

Comment:
This callout is likely confusing. Would love your take on it, and more than happy to go over it in detail on Slack or Zoom or here on GH, whichever you prefer.

Reply:
Perhaps we can focus less on restriction and more on inheritance. For example:

> The API key inherits the permissions of the authenticated user that creates or updates the API key.

Provided that this ^ statement is technically valid.

Reply:
I like inherit but it loses the following nuance: the API key's permissions are the intersection of the roles a user assigns to it and the user's permissions. "Inherit" seems to imply that the API key simply has the same permissions as the user it inherits from. What we should get across is the API key's permissions are "at most" those of the user.

"Limited by" is another phrase that came to mind but it clashes a bit with a statement further down:

> If you supply role descriptors, the API key's permissions will be the intersection of the supplied permissions and the authenticated user's current permissions, **thereby limiting the API key's access scope**.

I want to reserve using "limiting" to talking about the "assigned" role descriptors limiting the API key's permissions, and so went with "reserved" for the inverse case.

Reply:
I like the idea of a broader statement here. Essentially what I want to get across is that any call to this API may change the access scope of the API key even if an end-user does not specify any `role_descriptors`. I'll think about how I can keep make the statement broader and possibly move some of the context into the `role_descriptors` sub-section.